### PR TITLE
Feature/dotnet 8

### PR DESCRIPTION
**Pull request by TimPurdum**
- Adds `net8.0` target framework to Core project
- Updates all nuget packages to .NET 8
- Updates all samples and tests to .NET 8
- Adds new `Blazor Web App` sample
- Fixed a bug with race conditions and initializing the `AuthenticationManager` class

### CODE REVIEW

File: samples/dymaptic.GeoBlazor.Core.Sample.WebApp/Components/RenderModeSelector.razor [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        @{
+            switch (_configuredRenderMode)
+            {
+                case InteractiveWebAssemblyRenderMode _:
+                    <span>Page and components will be rendered in the browser</span>
+                    break;
+                case InteractiveServerRenderMode _:
+                    <span>Page and components will be rendered on the server and updated via SignalR</span>
+                    break;
+                case InteractiveAutoRenderMode _:
```
Comment (seahro) [inline samples/dymaptic.GeoBlazor.Core.Sample.WebApp/Components/RenderModeSelector.razor:20]: Ooo...some fresh new rendering options

File: samples/dymaptic.GeoBlazor.Core.Sample.WebApp/dymaptic.GeoBlazor.Core.Sample.WebApp.csproj [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+    <PropertyGroup>
+        <TargetFramework>net8.0</TargetFramework>
+        <Nullable>enable</Nullable>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <UserSecretsId>8fd6e776-4434-4649-b202-09b0c3b479f5</UserSecretsId>
```
Comment (seahro) [inline samples/dymaptic.GeoBlazor.Core.Sample.WebApp/dymaptic.GeoBlazor.Core.Sample.WebApp.csproj:7]: dont know if you wanted this here

File: samples/dymaptic.GeoBlazor.Core.Sample.ServerOAuth/dymaptic.GeoBlazor.Core.Sample.ServerOAuth.csproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
 
     <PropertyGroup>
-        <TargetFramework>net7.0</TargetFramework>
+        <TargetFramework>net8.0</TargetFramework>
         <Nullable>enable</Nullable>
         <ImplicitUsings>enable</ImplicitUsings>
         <UserSecretsId>8cc29e4d-db60-4432-bdd1-81b5fd050e1b</UserSecretsId>
```
Comment (seahro) [inline samples/dymaptic.GeoBlazor.Core.Sample.ServerOAuth/dymaptic.GeoBlazor.Core.Sample.ServerOAuth.csproj:7]: I am assuming this needs to be removed?

File: src/dymaptic.GeoBlazor.Core/Components/Renderers/Renderer.cs [hunk 4]
```diff
@@ -99,12 +99,21 @@ public enum RendererType
 #pragma warning restore CS1591
 }
 
+/// <summary>
```
Comment (seahro) [inline src/dymaptic.GeoBlazor.Core/Components/Renderers/Renderer.cs:102]: good catch

File: src/dymaptic.GeoBlazor.Core/ReadMe.md [hunk 5]
```diff
@@ -82,11 +84,26 @@ ArcGIS JavaScript API, but without having to write a single line of JavaScript.
...
+   var provider = new FileExtensionContentTypeProvider();
+   provider.Mappings[".wsv"] = "application/octet-stream";
+
+   app.UseStaticFiles();
+   // NOTE: for some reason, you still need the plain "UseStaticFiles" call above
+   app.UseStaticFiles(new StaticFileOptions
+   {
+       ContentTypeProvider = provider
+   });
+   ```
```
Comment (seahro) [inline src/dymaptic.GeoBlazor.Core/ReadMe.md:107]: Do we want to add a blurb about the licensing in this readme?

**Review by seahro — APPROVED** [review]
Couple comments.  Looks good

File: samples/dymaptic.GeoBlazor.Core.Sample.WebApp/dymaptic.GeoBlazor.Core.Sample.WebApp.csproj (hunk 2 above)
Comment (TimPurdum) [reply samples/dymaptic.GeoBlazor.Core.Sample.WebApp/dymaptic.GeoBlazor.Core.Sample.WebApp.csproj:7]: Yep, that's fine. It's a server-based AspNetCore executable. When I've asked you to remove them is when they are not in the Executable project, or when they are in Wasm which doesn't support user secrets.

File: samples/dymaptic.GeoBlazor.Core.Sample.ServerOAuth/dymaptic.GeoBlazor.Core.Sample.ServerOAuth.csproj (hunk 3 above)
Comment (TimPurdum) [reply samples/dymaptic.GeoBlazor.Core.Sample.ServerOAuth/dymaptic.GeoBlazor.Core.Sample.ServerOAuth.csproj:7]: removed?

File: src/dymaptic.GeoBlazor.Core/ReadMe.md (hunk 5 above)
Comment (TimPurdum) [reply src/dymaptic.GeoBlazor.Core/ReadMe.md:107]: The Registration logic is added on the Registration PR

**Pull request merged**